build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java

```diff
@@ -213,8 +213,8 @@ public void beforeStart() {
                 // if metrics were not enabled explicitly for gradlew run we should disable them
                 else if (node.getSettingKeys().contains("telemetry.metrics.enabled") == false) { // metrics
                     node.setting("telemetry.metrics.enabled", "false");
-                } else if (node.getSettingKeys().contains("telemetry.tracing.enabled") == false
-                    && node.getSettingKeys().contains("tracing.apm.enabled") == false) { // tracing
+                } else if ((node.getSettingKeys().contains("telemetry.tracing.enabled")
+                    || node.getSettingKeys().contains("tracing.apm.enabled")) == false) { // tracing
```

Comment:
I would prefer we remove the old setting being used here.
we should try not have deprecated settings used in our own codebase

Reply:
Done 👍

Reply:
I see this was requested.  I suggest moving this to a different PR and including a test.  This PR should focus on removing instrument.

Reply:
Moved to https://github.com/elastic/elasticsearch/pull/105076